src/cart/crt_hg.c

```diff
@@ -1021,10 +1021,14 @@ crt_hg_req_send_cb(const struct hg_cb_info *hg_cbinfo)
 				rpc_priv->crp_output_got = 1;
 				rc = rpc_priv->crp_reply_hdr.cch_rc;
 			} else {
-				RPC_ERROR(rpc_priv,
-					  "HG_Get_output failed, hg_ret: %d\n",
-					  hg_ret);
-				rc = -DER_HG;
+				if (hg_ret != HG_NOMEM) {
+					RPC_ERROR(rpc_priv,
+						  "HG_Get_output failed, "
+						  "hg_ret: %d\n", hg_ret);
+					rc = -DER_HG;
+				} else {
+					rc = -DER_NOMEM;
+				}
```

Comment:
My only concern is that we are making this change once, buy a quick grep shows there are 47 cases where we set rc to -DER_HG and 194 where we return -DER_HG, and many of those should probably also be treated like this, but just aren't triggered currently by the fault injection testing.

Outside the scope of this PR but I think it's something we should look at at some point.